File: app/frontend/src/api/template.ts
import request from "./index";
import type { ApiResponse } from "./types";

// 模板数据结构
export interface Template {
  id?: string;
  name: string;
  width: number;
  height: number;
  scale: number;
  imagePath: string;
  // API 返回的字段
  offset_x?: number;
  offset_y?: number;
  // 兼容旧字段（本地存储使用）
  offsetTop?: number;
  offsetRight?: number;
  offsetBottom?: number;
  offsetLeft?: number;
  url?: string;
  createdAt?: string;
}

// 创建模板请求参数
export interface CreateTemplateRequest {
  name: string;
  width: number;
  height: number;
  scale: number;
  imagePath: string;
  offset_x: number;
  offset_y: number;
  url: string;
}

// 创建模板
export function apiCreateTemplate(
  data: CreateTemplateRequest,
): Promise<ApiResponse<Template>> {
  return request.post<ApiResponse<Template>>("/v1/templates", data);
}

// 获取模板列表
export function apiGetTemplates(): Promise<ApiResponse<Template[]>> {
  return request.get<ApiResponse<Template[]>>("/v1/templates");
}

// 删除模板
export function apiDeleteTemplate(id: string): Promise<ApiResponse<void>> {
  return request.delete<ApiResponse<void>>(`/v1/templates/${id}`);
}

// 更新模板
export function apiUpdateTemplate(
  id: string,
  data: CreateTemplateRequest,
): Promise<ApiResponse<Template>> {
  return request.put<ApiResponse<Template>>(`/v1/templates/${id}`, data);
}

export default {
  apiCreateTemplate,
  apiGetTemplates,
  apiDeleteTemplate,
  apiUpdateTemplate,
};

File: backend/models/template.go
package models

// Template 模版模型
type Template struct {
	Model
	UserID  uint    `gorm:"not null;index" json:"user_id"`      // 用户ID
	Name    string  `gorm:"size:100;not null" json:"name"`      // 模版名称
	URL     string  `gorm:"size:500;not null" json:"url"`       // 模版URL
	Width   int     `gorm:"not null;default:0" json:"width"`    // 素材宽度
	Height  int     `gorm:"not null;default:0" json:"height"`   // 素材高度
	OffsetX float64 `gorm:"not null;default:0" json:"offset_x"` // X轴偏移
	OffsetY float64 `gorm:"not null;default:0" json:"offset_y"` // Y轴偏移
	Scale    float64 `gorm:"not null;default:1" json:"scale"`      // 缩放比例
	Rotation float64 `gorm:"not null;default:0" json:"rotation"`   // 旋转角度（度数，顺时针）
}

func (Template) TableName() string {
	return "templates"
}
